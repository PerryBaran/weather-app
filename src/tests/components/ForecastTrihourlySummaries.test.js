import React from "react";
import { render } from "@testing-library/react";
import ForecastTrihourlySummaries from "../../components/ForecastTrihourlySummaries";

describe("ForecastTrihourlySummaries", () => {
  const validProps = {
    forecasts: [
      {
        dateTime: 123,
        time: "now",
        icon: "200",
        temp: 12,
        maxTemp: 13,
        minTemp: 11,
        humidity: 44,
        weather: "really cold",
        windSpeed: 66,
      },
      {
        dateTime: 456,
        time: "later",
        icon: "400",
        temp: 22,
        maxTemp: 23,
        minTemp: 21,
        humidity: 45,
        weather: "less cold",
        windSpeed: 99,
      },
    ],
  };

  describe("snapshot", () => {
    it("renders correctlty", () => {
      const { asFragment } = render(
        <ForecastTrihourlySummaries forecasts={validProps.forecasts} />
      );

      expect(asFragment()).toMatchSnapshot();
    });
  });

  describe("list", () => {
    it("renders correct values for props", () => {
      const { getAllByTestId } = render(
        <ForecastTrihourlySummaries forecasts={validProps.forecasts} />
      );

      expect(getAllByTestId("forecast-trihourly-summary")).toHaveLength(8);
    });
  });
});
