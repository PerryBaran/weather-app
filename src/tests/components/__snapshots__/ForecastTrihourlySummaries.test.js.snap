// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastTrihourlySummaries snapshot renders correctlty 1`] = `
<DocumentFragment>
  <div
    class="forecast-trihourly-summaries"
  >
    <div
      class="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p
        class="forecast-trihourly-summary__time"
      >
        00:00
      </p>
      <p
        class="forecast-trihourly-summary__no-data"
      >
        No 
        <br />
        Data
      </p>
    </div>
    <div
      class="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p
        class="forecast-trihourly-summary__time"
      >
        03:00
      </p>
      <p
        class="forecast-trihourly-summary__no-data"
      >
        No 
        <br />
        Data
      </p>
    </div>
    <div
      class="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p
        class="forecast-trihourly-summary__time"
      >
        06:00
      </p>
      <p
        class="forecast-trihourly-summary__no-data"
      >
        No 
        <br />
        Data
      </p>
    </div>
    <div
      class="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p
        class="forecast-trihourly-summary__time"
      >
        09:00
      </p>
      <p
        class="forecast-trihourly-summary__no-data"
      >
        No 
        <br />
        Data
      </p>
    </div>
    <div
      class="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p
        class="forecast-trihourly-summary__time"
      >
        12:00
      </p>
      <p
        class="forecast-trihourly-summary__no-data"
      >
        No 
        <br />
        Data
      </p>
    </div>
    <div
      class="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p
        class="forecast-trihourly-summary__time"
      >
        15:00
      </p>
      <p
        class="forecast-trihourly-summary__no-data"
      >
        No 
        <br />
        Data
      </p>
    </div>
    <div
      class="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p
        class="forecast-trihourly-summary__time"
      >
        18:00
      </p>
      <p
        class="forecast-trihourly-summary__no-data"
      >
        No 
        <br />
        Data
      </p>
    </div>
    <div
      class="forecast-trihourly-summary"
      data-testid="forecast-trihourly-summary"
    >
      <p
        class="forecast-trihourly-summary__time"
      >
        21:00
      </p>
      <p
        class="forecast-trihourly-summary__no-data"
      >
        No 
        <br />
        Data
      </p>
    </div>
  </div>
</DocumentFragment>
`;
